diff --git a/atreugo.go b/atreugo.go
index c81bce9..87dc2ce 100644
--- a/atreugo.go
+++ b/atreugo.go
@@ -121,6 +121,13 @@ func newFasthttpServer(cfg Config) *fasthttp.Server {
 func (s *Atreugo) handler() fasthttp.RequestHandler {
 	handler := s.router.Handler

+	if s.cfg.BeforeHandler != nil {
+		handler = func(ctx *fasthttp.RequestCtx) {
+			s.cfg.BeforeHandler(ctx)
+			s.router.Handler(ctx)
+		}
+	}
+
 	if len(s.virtualHosts) > 0 {
 		handler = func(ctx *fasthttp.RequestCtx) {
 			hostname := strconv.B2S(ctx.URI().Host())
diff --git a/types.go b/types.go
index 5265ec7..fce8b7d 100644
--- a/types.go
+++ b/types.go
@@ -369,6 +369,9 @@ type Config struct { // nolint:maligned
 	//
 	// NetHttpFormValueFunc gives a FormValueFunc func implementation that is consistent with net/http.
 	FormValueFunc fasthttp.FormValueFunc
+
+	// BeforeHandler Function to modify ctx before starting path handling.
+	BeforeHandler func(ctx *fasthttp.RequestCtx)
 }

 // StaticFS represents settings for serving static files
